pytorch_lightning/core/lightning.py

```diff
@@ -1998,3 +1998,10 @@ def get_from_queue(self, queue: torch.multiprocessing.SimpleQueue) -> None:
         self.trainer.callback_metrics.update(
             apply_to_collection(callback_metrics, np.ndarray, lambda x: torch.tensor(x))
         )
+
+    def __getstate__(self) -> Dict[str, Any]:
+        # prevent copying data
+        skipped_keys = {
+            "trainer", "train_dataloader", "val_dataloader", "test_dataloader", "predict_dataloader", "datamodule"
```

Comment:
Why do you skip the dataloader ones which are functions, not properties or attributes?

Reply:
Just guessing, but if the user provides the loaders to the trainer, these functions are patched with callable classes also containing the loader object.

Reply:
This doesn't play nice with ddp_spawn and ddp_cpu. I needed to make this optional through a context manager.